backend/src/routes/analytics.py

```
@@ -0,0 +1,53 @@
+        success_rate = (success_trades / total_trades * 100) if total_trades > 0 else 0
+
+        # Get active positions
+        active_tokens = len(auto_trader.owned_tokens)
+
+        # Today's profit
+        now = datetime.now()
+        today = now.date()
+        today_profit = sum(t.get('profit_usd', 0) for t in history if datetime.fromisoformat(t['sell_time']).date() == today)
+
+        # Weekly profit
+        one_week_ago = now - timedelta(days=7)
```

Ruff (0.14.14)

[error] Undefined name `timedelta`

(F821)